Reset Password Component

mohamedsgap opened the pull request:
Just finished creating ResetPassword component, with 3 test specs! #27 
![Screenshot from 2020-09-02 06-51-48](https://user-images.githubusercontent.com/30293804/91976798-d4e98980-ed21-11ea-9da5-49d6cd2a2045.png)

mohamedsgap committed:
chore: install react-final-form

mohamedsgap committed:
chore: install final-form

mohamedsgap committed:
feat: built ResetPassword component

mohamedsgap committed:
style: styled ResetPassword

mohamedsgap committed:
write test specs

mohamedsgap committed:
add test snapshots

mohamedsgap committed:
fix Codacy #1

mohamedsgap commented:
@AbdullahMuhammed5 @aashrafh @mwissem guys, we need to update `Codacy` rules, to be compatible with React code!, I think @mnourrhan talked about this before!

AbdullahMuhammed5 commented on src/containers/ResetPassword/ResetPassword.js:
```diff
@@ -0,0 +1,146 @@
+          }}
+          render={({ handleSubmit, form, values }) => (
+            <form onSubmit={handleSubmit} className={classes.form}>
+              <Field name="password">
```
TextField component is more elegant, and by using it you will follow my layout and I will send you a screen shoot from my work.

aashrafh replied:
```Field``` is the same markup used in the final form documentation

AbdullahMuhammed5 reviewed (commented):
Great work. I have some comment would you mind to take a look at it.

This how my component looks like: 
![Screenshot from 2020-09-02 15-31-51](https://user-images.githubusercontent.com/29069500/91990405-f94d6200-ed31-11ea-9c96-1c2177c95a85.png)

aashrafh commented on src/containers/ResetPassword/ResetPassword.js:
```diff
@@ -0,0 +1,146 @@
+        </Typography>
+        <Form
+          onSubmit={onSubmit}
+          validate={(values) => {
```
I think it will be more readable if you separated the logic to a function like onSubmit

aashrafh commented on src/containers/ResetPassword/ResetPassword.js:
```diff
@@ -0,0 +1,146 @@
+            }
+            return errors;
+          }}
+          render={({ handleSubmit, form, values }) => (
```
Same previous note here. Make it more readable.

aashrafh commented on src/containers/ResetPassword/ResetPassword.js:
```diff
@@ -0,0 +1,146 @@
+  );
+}
+
+function Copyright() {
```
Why not separate this component into a separate file?

AbdullahMuhammed5 replied:
Yes i have the component in the forget password and register branch you can get it use it.

mohamedsgap commented:
@AbdullahMuhammed5 @aashrafh for the form validation, I used react-final- form! It validate the input fields very well, May for how Input fields look like I can use TextField component from MUI!! 
So what u think ??

And one final thing regarding Coday, it doesn't like Hooks, and all strings must be double quotes, also we need to update these rules!

AbdullahMuhammed5 commented:
> @AbdullahMuhammed5 @aashrafh for the form validation, I used react-final- form! It validate the input fields very well, May for how Input fields look like I can use TextField component from MUI!!
> So what u think ??
> 
> And one final thing regarding Coday, it doesn't like Hooks, and all strings must be double quotes, also we need to update these rules!

First I don't see that we need a validation package. I made the register and forget password manually. You can do so they are just two fields.

regarding Codacy I don't have any suggestions, and i don't know if it is legal to overwrite or disable some of their rules.

mohamedsgap committed:
chore: remove react-final-form

mohamedsgap committed:
simplified ResetPassword without react-final-form

mohamedsgap committed:
fix a typo

mohamedsgap committed:
update test specs with snapshots

mohamedsgap committed:
fix Codacy issue: by using style MUI hook at the top

mohamedsgap commented:
@AbdullahMuhammed5 @mwissem @aashrafh  just improved the validation manually of the from without react-final-form, so I removed it, also used TextField instead of Field. Update tests with snapshots, and fixed Codacy issue :)

aashrafh commented on src/containers/ResetPassword/ResetPassword.js:
```diff
@@ -0,0 +1,165 @@
+  );
+}
+
+function Copyright() {
```
Repeated Component.

mohamedsgap replied:
I made it on purpose, because you already change this component by removing the like it MUI site!, it will cause conflict when we merge, I gonna import once all the current PRs got merged @aashrafh

aashrafh commented on src/containers/ResetPassword/ResetPassword.test.js:
```diff
@@ -0,0 +1,14 @@
+  expect(wrapper).toMatchSnapshot();
+});
+
+it("should render TextFiels correctly", () => {
```
One snapshot for the whole component is enough

mohamedsgap replied:
I really do it, so when I interact with api, I need to spy on the both functions of button and form, so shallow the TextField gonna save much headache 😁 @aashrafh

aashrafh reviewed (changes requested):
Very Good. Just see the notes on the code

AbdullahMuhammed5 commented on src/containers/ResetPassword/__snapshots__/ResetPassword.test.js.snap:
```diff
@@ -0,0 +1,99 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
I believe this file should be ignored in git.

mohamedsgap replied:
the same thing I told to @aashrafh  "I really do it, so when I interact with api, I need to spy on the both functions of button and form, so shallow and mount on  the TextField gonna save much headache"" @AbdullahMuhammed5
